src/main/java/org/checkerframework/checker/dividebyzero/DivByZeroTransfer.java

```diff
@@ -78,33 +78,63 @@ enum BinaryOperator {
 private AnnotationMirror refineLhsOfComparison(
     Comparison operator, AnnotationMirror lhs, AnnotationMirror rhs) {
 
+  // Handle "not equal" comparison
+  	//top NE 0 then top is nonzero
     if (operator == Comparison.NE) {
+        // If rhs is zero, lhs must be NonZero
         if (equal(rhs, reflect(Zero.class))) {
             return reflect(NonZero.class); // lhs cannot be Zero if it is not equal to Zero
         }
-    } else if (operator == Comparison.EQ) {
+    } 
+    // Handle "equal" comparison
+    //top is equal to zero then top is zero
+    else if (operator == Comparison.EQ) {
+        // If rhs is zero, refine lhs to Zero
         if (equal(rhs, reflect(Zero.class))) {
-            return reflect(Zero.class); // Refine lhs to Zero
+            return reflect(Zero.class); 
```

Comment:
Why only do refinement when the comparison is against zero, and only when the it's the RHS that is zero? I think you can write a more general rule. 

For example, you should be able to handle `if (0 == x)` just as well as `if (x == 0)`. And, it would be good if your analysis doesn't warn against code like this:
```
if (x == 1) { int y = 5/x; }
```
I think right now you will warn about this, because this refinement code won't be triggered unless the comparison is against zero. By contrast, I think you would handle `if (x != 0) { ... }`, which users will probably find confusing.